BlakeDonn/rancid-tomatillos
# Tests

### What's this PR do?  
- This PR adds initial testing to the Card component
- This PR adds conditional rendering to the MainPage and login link
- This PR adds error handling for both fetch requests
- This PR re-organizes our file structure to better separate components

### Where should the reviewer start?  
- Start by checking out the new switch statement logic to get a better understanding of how the different state effects the rendering

### How should this be manually tested?  
- Adjust API calls for bad tests and check the error handlers are working correctly
- Check that you can log out when all-ready logged in.

### Any background context you want to provide?  
- We worked on all of these changes together, and agree the error handling and conditional rendering needs polish 

### What are the relevant tickets?  
closes #42 closes #43 closes #41 closes #40 closes #30 closes #26 closes #25 closes #22 closes closes #9 closes   #5
   #44
### Screenshots (if appropriate)  
### Questions:
@hannakim91  Thanks for keeping up on the issue cards, looks great!
